### DISCUSSION TITLE
Update Frequency of Values AdoraWash V2000

### DISCUSSION THREAD

**Discussion by Generalhassan** [siku2/hass-vzug#51] [Device Support]
Hi,
First of all great integration, thanks for your work :)
In my logbook of the integration i am seeing that some sensor states change every 2 minutes (even if i am not using the machine) and i see that the Wifi light is blinking constantly on the machine itself.

Maybe this is a really basic question but how do i configure the integration so that it does not constantly pull data from the washing machine - or is this something related to the V Zug account? However, for the dryer which i have not yet connected to Home Assitant it seems to work correctly in a way that the device is fully off and is only reachable in my V Zug app when i actually use it (which is the way it should be)

**Comment by siku2** [comment]
The update interval is currently hard-coded here: https://github.com/siku2/hass-vzug/blob/24f8c2e7a646e588fef3380cba68dc7d027deb8c/custom_components/vzug/shared.py#L49-L69

I'm not sure what you're trying to achieve. In order to tell if a device is active we have to actively make requests to it.

If you have another way to tell whether the device is active and you want to disable polling when it's not active, I would recommend using the Spook: https://github.com/frenck/spook

Spook has a service `homeassistant.disable_polling` that can be used to disable the polling on an entire config entry (i.e. device).

**Comment by Generalhassan** [comment]
Thanks for your quick reply and the information regarding the refresh timing. I was just trying to figure out why the Wifi light on the washing machine was constantly blinking and why it was not going into a proper sleep mode where it is not reachable via wifi (like my dryer does which i have not connected to the plugin/HA yet) purely to save energy :)

**Reply by siku2** [reply]
That's a fair question. I must admit I didn't notice this myself.
As far as I know the integration isn't actively doing anything to force the devices to remain awake. However, it's very possible that the periodic polling itself is what's keeping the device from entering a sleep state.
Can you confirm that if the integration is disabled the device does behave as you would expect? (i.e. the wifi ind. stops blinking and the device uses less power)

**Reply by Generalhassan** [reply]
First of all thanks for your reply and sorry for me taking so long to look into this. I finally disable the integration and indeed the washing machine now behaves as expect (no wifi light blinking when not in use but still getting the notification on the V Zug app). To be honest i did not measure the energy consumption with the integration on vs off so i don`t know how much of a difference it makes but once i get a plug with consumption measurement i will post it here just for information :)

**Comment by Urseen** [comment]
I have the same problem with my AdoraWaschen4000. As long as the integration was active, the machine made a quiet but clearly audible “click” at short intervals and the Wi-Fi symbol turned on. Without the integration, this clicking stops.

**Comment by Soardiac** [comment]
Hey, I'll add a PR to "fix" this, once #149 and #150 are done. Well, not really a _fix_ since nothing is _broken_. But I implemented a different polling strategy locally that keeps the appliance asleep much longer, hopefully without losing functionality.
As of now the integration behaves like an always-open V-ZUG app: it polls for everything every 30s. The first call wakes the appliance (returns error 503.11), then there is a ~22s window for successful polls before it falls asleep again. So at 30s the appliance is awake ~22s, sleeps ~8s, rinse and repeat. That's roughly 15 hours a day awake for a machine that is doing nothing.

While looking into #146, what my V6000 actually does and what's in the HA log, I found that `getDeviceStatus` is what wakes it up, while `getLastPUSHNotifications` doesn't wake it at all and is still up to date. So I can poll `getLastPUSHNotifications` every 60s to detect program ends in almost real time without wakeing the appliance, and do one full poll when something actually happened. `getDeviceStatus` then only runs every 5 or 10 minutes, just to catch a program someone started by hand.
This way, my appliance gets much more sleep (22-23h on a totally inactive day) and I still don't really lose information. While a program is running the appliance stays awake anyway, so polling during a run costs nothing and the countdown works as before. And a _timed_ start does send a push notification, so the case where nobody is standing at the machine is covered. Exactly what the app is doing, btw: no push for manual starts but one for timed starts. 
What's left is the manual start: if let's say my wife starts it, HA might not notice for a few minutes, but still well before it ends. For me that's hypothetical, I have no HA use case that needs those first minutes. But maybe someone does? Would that work for you, too, @Generalhassan and @Urseen?

**Reply by Generalhassan** [reply]
That sounds great and would definitely work for me. Thanks for coming back to this issue and finding a solution - in the meantime i used AI to fix it locally (i can only vibe code haha) but it sounds like your solution is much better so i appreciate a proper fix! 

This is what i had done:

As a stopgap I patched `shared.py` locally so the polling interval follows the `Inactive` flag: 30s when active, 50 minutes when inactive. It works — my AdoraWash V2000 and AdoraDry V2000 went from ~120 state polls/hour each to about 11 per device overnight, and the WiFi LEDs are finally dark. But it's crude compared to what you're describing, since it just slows everything down instead of separating the calls that actually wake the appliance.

Your `getLastPUSHNotifications` finding is the piece I was missing, and my logs confirm it on the V2000 too: in the same poll cycle, push notifications came back in 186ms with current data while `getDeviceStatus` threw a 503 and needed ~6.5s with the retry.

Re manual starts — not a concern for me. I only need the end-of-program notification, so a few minutes of delay at the start is a fair trade for 22-23h of sleep. Looking forward to the PR.

**Reply by Soardiac** [reply]
Siku2 has merged https://github.com/siku2/hass-vzug/pull/152 into main, so reduced polling a.k.a. longer sleep time is there. There is no new release, but you can test by using the files from main locally in your HA. You'll get all the other changes/diffs, too of course. So maybe you'll check all the [commits in main](https://github.com/siku2/hass-vzug/commits/main/) since your last update before you copy...
There is one more open PR #153 around this to add a refresh-button should a user or an automation want to instantly refresh (because of reduced polling)
